code/modules/mob/living/carbon/human/species.dm

```diff
@@ -402,8 +402,20 @@ GLOBAL_LIST_EMPTY(features_by_species)
 			if(I)
 				I.Remove(C)
 				QDEL_NULL(I)
-
-	for(var/organ_path in mutant_organs)
+	for(var/external_organ in C.external_organs)
+		// External organ checking. We need to check the external organs owned by the carbon itself,
+		// because we want to also remove ones not shared by its species.
+		// This should be done even if species was not changed.
+		if(external_organ in external_organs)
+			continue // Don't remove external organs this species is supposed to have.
+		var/obj/item/organ/I = C.getorgan(external_organ)
```

Comment:
use a better variable name here, `I` is not good enough

Reply:
I have changed this to "current_organ" both in this and the section that handles mutant organs.